src/rofiles-fuse/main.c

```diff
@@ -231,7 +231,7 @@ callback_chmod (const char *path, mode_t mode)
 {
   path = ENSURE_RELPATH (path);
   VERIFY_WRITE(path);
-  if (fchmodat (basefd, path, mode, 0) != 0)
+  if (fchmodat (basefd, path, mode, AT_SYMLINK_NOFOLLOW) != 0)
```

Comment:
This is tricky; not all `*at()` ops support this flag.  In particular `man-pages-4.09-3.fc26.noarch` says:

>        AT_SYMLINK_NOFOLLOW
>              If pathname is a symbolic link, do not dereference it: instead operate on the link itself.  This flag is not currently implemented.

There was a thread about this recently...

https://marc.info/?l=linux-kernel&m=148830147803162&w=2
https://marc.info/?l=linux-fsdevel&m=149193779929561&w=2

Reply:
So we probably need to simulate this via `openat(...O_NOFOLLOW)`, ignore `ELOOP` (meaning symlinks), and for regular files use `fchmod()` on the fd.

Reply:
I think that is overdoing it. Currently (as in the commit message) the kernel will never emit the fuse chmod operation on a symlink, because the kernel does not support it. If, later the kernel starts implementing this it might. However, if it does, then the fchmodat call is likely to work, so the code as it will work.

Reply:
But today the glibc syscall wrapper returns `EINVAL` if any flags are set.

I wrote https://github.com/ostreedev/ostree/pull/1141 and verified that this patch causes the tests to fail.

Reply:
So let's drop the flag and add a comment with links to the above linux-fsdevel threads?